pkg/gpg/gpg_forwarding.go

```diff
@@ -0,0 +1,126 @@
+package gpg
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type GPGConf struct {
+	PublicKey  []byte
+	OwnerTrust []byte
+	SocketPath string
+}
+
+func (g *GPGConf) StopGpgAgent() error {
+	return exec.Command("gpgconf", []string{"--kill", "gpg-agent"}...).Run()
+}
+
+func (g *GPGConf) ImportGpgKey() error {
+	gpgImportCmd := exec.Command("gpg", "--import")
+
+	stdin, err := gpgImportCmd.StdinPipe()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		defer stdin.Close()
+		_, _ = stdin.Write(g.PublicKey)
+	}()
+
+	return gpgImportCmd.Run()
+}
+
+func (g *GPGConf) ImportOwnerTrust() error {
+	gpgOwnerTrustCmd := exec.Command("gpg", "--import-ownertrust")
+
+	stdin, err := gpgOwnerTrustCmd.StdinPipe()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		defer stdin.Close()
+		_, _ = stdin.Write(g.OwnerTrust)
+	}()
+
+	return gpgOwnerTrustCmd.Run()
+}
+
+func (g *GPGConf) SetupGpgConf() error {
+	_, err := os.Stat(filepath.Join(os.Getenv("HOME"), ".gnupg", "gpg.conf"))
+	if err != nil {
+		_, err = os.Create(filepath.Join(os.Getenv("HOME"), ".gnupg", "gpg.conf"))
+		if err != nil {
+			return err
+		}
+	}
+
+	gpgConfig, err := os.ReadFile(filepath.Join(os.Getenv("HOME"), ".gnupg", "gpg.conf"))
```

Comment:
maybe we can have a method `configPath()` returning `filepath.Join(os.Getenv("HOME"), ".gnupg", "gpg.conf")`

Reply:
Done :+1: